Documentation/manpages/gsc.rst

```diff
@@ -0,0 +1,239 @@
+
+**Configuration**
+
+GSC is configured via a configuration file called ``config.json`` with the
+following parameters.
+
+   .. option:: distro
+
+      Defines Linux distribution to be used to build Graphene in. Currently
+      supported values are ``ubuntu18.04``/``ubuntu16.04``.
+
+   .. option:: graphene_repository
+
+      Source repository of Graphene. Default value:
+      https://github.com/oscarlab/graphene
+
+   .. option:: graphene_branch
+
+      Branch of the ``graphene_repository``. Default value: master
+
+   .. option:: sgxdriver_repository
+
+      Source repository of the Intel SGX driver. Default value:
+      https://github.com/01org/linux-sgx-driver.git
+
+   .. option:: sgxdriver_branch
+
+      Branch of the ``sgxdriver_repository``. Default value: sgx_driver_1.9
+
+Run graphenized Docker images
+=============================
+
+Execute Docker run command via Docker CLI and provide gsgx and isgx/sgx device,
+and the PSW/AESM socket. Additional Docker options and application arguments may
+be supplied to the Docker run command.
+
+.. program:: docker
+
+:command:`docker` run --device=/dev/gsgx --device=/dev/isgx -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket [*OPTIONS*] gsc-<*IMAGE-NAME*>[:<*TAG*>] [<*APPLICATION-ARGUMENTS*>]
+
+   .. option:: IMAGE-NAME
+
+      Name of original image (without GSC build).
+
+   .. option:: TAG
+
+      Tag of the image to be used.
+
+   .. option:: APPLICATION-ARGUMENTS
+
+      Application arguments to be supplied to the application launching inside
+      the Docker container and Graphene.
+
+   .. option:: OPTIONS
+
+      Docker run options. Common options include ``-it`` (interactive with
+      terminal) or ``-d`` (detached). Please see
+      `Docker manual <https://docs.docker.com/engine/reference/commandline/run/>`__
+      for details.
+
+
+Execute with Linux PAL instead of Linux-SGX PAL
+-----------------------------------------------
+
+When specifying ``-L`` during GSC ``build``, you may select the Linux PAL at
+Docker run time instead of the Linux-SGX PAL by specifying the environment
+variable ``LINUX_PAL`` as an option to the Docker ``run`` command.
+
+.. code-block:: bash
+
+    docker run ... --env LINUX_PAL=linux gsc-<image-name> ...
+
+Example
+=======
+
+This example shows how to graphenize the public Docker image of Python3. This
+example assumes that all prerequisites are installed and configured. For more
+examples refer to the test folder of ``gsc``.
+
+1) Pull public Python image from Dockerhub:
+
+.. code-block:: bash
+
+   docker pull python
+
+2) Graphenize the Python image using ``gsc``:
+
+.. code-block:: bash
+
+   cd Tools/gsc
+   gsc build python test/ubuntu18.04-python3.manifest
+
+3) Test the graphenized Docker image:
+
+.. code-block:: bash
+
+   docker run --device=/dev/gsgx --device=/dev/isgx -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket python -c 'print("HelloWorld!")'
```

Comment:
This command didn't work for me. I had to specify the `gsc_python` image name in above command.

`docker run --device=/dev/gsgx --device=/dev/isgx -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket gsc-python -c 'print("HelloWorld!")'`

Reply:
The above command took almost 1min 44 sec to print Helloworld.  I checked with `time <command>`. I tried running graphene SGX in Azure Cloud Computing VM.